packages/react-components/src/components/ChatMessage/MyMessageComponents/ChatMessageComponentAsEditBoxPicker.tsx

```diff
@@ -44,7 +46,8 @@ export type ChatMessageComponentAsEditBoxPickerProps = {
   strings: MessageThreadStrings;
   /* @conditional-compile-remove(mention) */
   mentionLookupOptions?: MentionLookupOptions;
-  richTextEditor?: boolean;
+  /* @conditional-compile-remove(rich-text-editor) */
+  richTextEditor?: boolean | /* @conditional-compile-remove(rich-text-editor-image-upload) */ RichTextEditorOptions;
```

Comment:
Looks like we could just replace the boolean here rather than create the union of types?

Reply:
This is for consistency with other APIs

<img width="912" alt="image" src="https://github.com/Azure/communication-ui-library/assets/73612854/ce31bb08-2d60-4c4b-a86a-27fdce1635da">

<img width="601" alt="image" src="https://github.com/Azure/communication-ui-library/assets/73612854/b11c8750-a219-4676-b484-5cbef41d8166">

Reply:
I believe those were made as the original API had been GA'd. We should avoid this pattern if possible.